File: Python/Resistor Color.py


def color_code(color, lookup=1):
    dict = colors(lookup)
    return dict[color.capitalize()]


def colors(lookup=0):
    dict = { "Black":0,
             "Brown":1,
             "Red":2,
             "Orange":3,
             "Yellow":4,
             "Green":5,
             "Blue":6,
             "Violet":7,
             "Grey":8,
             "White":9 }
    if lookup == 0: return list(dict.keys())
    else: return dict

print(color_code("orange"))

File: C#/Resistor Color.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class ResistorColor
{
    public static Dictionary<int, string> ColorDict = new Dictionary<int, string>()
    {
        {0, "black"},
        {1, "brown"},
        {2, "red"},
        {3, "orange"},
        {4, "yellow"},
        {5, "green"},
        {6, "blue"},
        {7, "violet"},
        {8, "grey"},
        {9, "white"}
    };

    public static int ColorCode(string color) => ColorDict.First(val => val.Value == color).Key;

    public static string[] Colors() => ColorDict.Values.ToArray();
}